src/sortStudents.ts

```diff
@@ -1,16 +1,63 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+const getAverageGrade = (grades: number[]) : number => {
+  return grades.reduce((previous, current) => previous + current
+    , 0) / grades.length;
+};
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const cloneStudents = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      return order === 'asc'
+        ? cloneStudents
+          .sort((stud1, stud2) => stud1[sortBy].localeCompare(stud2[sortBy]))
```

Comment:
Better give a more obvious name all over the code.
```suggestion
          .sort((student1, student2) => student1[sortBy].localeCompare(student2[sortBy]))
```